persistence/src/main/scala/com/lightbend/lagom/internal/persistence/cluster/ClusterDistribution.scala

```diff
@@ -0,0 +1,128 @@
+  ensureActiveInterval:    FiniteDuration
+)
+
+object ClusterDistributionSettings {
+  def apply(system: ActorSystem): ClusterDistributionSettings = {
+    val clusterShardingSettings = ClusterShardingSettings(system)
+    val config = system.settings.config.getConfig("akka.cluster.distribution")
+    ClusterDistributionSettings(config, clusterShardingSettings)
+  }
+
+  def apply(config: Config, clusterShardingSettings: ClusterShardingSettings): ClusterDistributionSettings = {
+    val ensureActiveInterval = config.getDuration("ensure-active-interval", TimeUnit.MILLISECONDS).milliseconds
+    ClusterDistributionSettings(clusterShardingSettings, ensureActiveInterval)
+  }
+}
+
+object ClusterDistribution extends ExtensionId[ClusterDistribution] with ExtensionIdProvider {
+
+  override def createExtension(system: ExtendedActorSystem): ClusterDistribution =
+    new ClusterDistribution(system)
+
+  override def lookup = ClusterDistribution
+
+  /**
+   * Sent to each entity in a cluster distribution to ensure it's active.
+   */
+  case class EnsureActive(entityId: EntityId)
+}
+
+/**
+ * Distributes a static list of entities evenly over a cluster, ensuring that they are all continuously active.
+ *
+ * This uses cluster sharding underneath, using a scheduled tick sent to each entity as a mechanism to ensure they are
+ * active.
+ *
+ * Entities are cluster sharding entities, so they can discover their ID by inspecting their name. Additionally,
+ * entities should handle the [[com.lightbend.lagom.internal.persistence.cluster.ClusterDistribution.EnsureActive]]
+ * message, typically they can do nothing in response to it.
+ */
+class ClusterDistribution(system: ExtendedActorSystem) extends Extension {
+
+  /**
+   * Start a cluster distribution.
+   *
+   * @param typeName The name of the type of entity. This is used as the cluster sharding type name.
+   * @param entityProps The props for the entity actor.
+   * @param entityIds The entity ids to distribute over the cluster.
+   * @param settings The cluster distribution settings.
+   * @return the actor ref of the [[ShardRegion]] that is to be responsible for the shard
+   */
+  def start(
+    typeName:    String,
+    entityProps: Props,
+    entityIds:   Seq[EntityId],
+    settings:    ClusterDistributionSettings
+  ): ActorRef = {
+
+    val extractEntityId: ShardRegion.ExtractEntityId = {
+      case msg @ EnsureActive(entityId) => (entityId, msg)
+    }
+    val extractShardId: ShardRegion.ExtractShardId = {
+      case EnsureActive(entityId) => entityId
+    }
+
+    val shardRegion = ClusterSharding(system).start(typeName, entityProps, settings.clusterShardingSettings,
+      extractEntityId, extractShardId)
+
+    system.systemActorOf(EnsureActiveActor.props(entityIds, shardRegion, settings.ensureActiveInterval), "cluster-disstribution-" + URLEncoder.encode(typeName, "utf-8"))
+
+    shardRegion
+  }
+
+}
+
+private[cluster] object EnsureActiveActor {
+  case object Tick
```

Comment:
It's only for reducing logging noise. Good to use it on scheduled messages since there is a chance they arrive after the actor is stopped.